### PR TITLE
Comments 2

### PR DESCRIPTION
**Pull request by ZanCorDX**
## 📝 Summary

Improving comments.

## 💡 Motivation and Context

I'm a really nice guy and I want people to enjoy the rbuilder.

## ✅ I have completed the following steps:

* [X] Run `make lint`
* [X] Run `make test`
* [ ] Added tests (if applicable)

### CODE REVIEW

File: crates/rbuilder/src/live_builder/order_input/clean_orderpool.rs [hunk 1]
```diff
@@ -19,6 +19,8 @@ use tokio::task::JoinHandle;
 use tokio_util::sync::CancellationToken;
 use tracing::{debug, error, info};
 
+/// Performs maintenance operations on every new header by calling OrderPool::head_updated.
+/// Also calls some metrics.
```
Comment (metachris) [inline crates/rbuilder/src/live_builder/order_input/clean_orderpool.rs]: what does "call metrics" mean? 🤔

File: crates/rbuilder/src/live_builder/simulation/mod.rs [hunk 2]
```diff
@@ -29,275 +25,46 @@ pub struct SlotOrderSimResults {
...
-/// Runner of the simulations.
-/// Create and call run()
-/// 1- Gets the orders orders/cancellations from new_order_sub
-/// 2- Using a SisTree checks when the orders re ready for simulation (nonce problems) and simulates them
-/// 3- Send simulation results via sim_results_receiver
-/// Also handles ShareBundle version replacements or cancellations (maybe this logic could be moved to an external object?):
-/// - Always send simulations with increasing `sequence_number`
-/// - When replacing an order we always send a cancellation for the previous one.
-/// - When we gat a cancellation we propagate it and never again send an update.
-impl<DB: Database + Clone + Send + 'static> SimulationJob<DB> {
```
Comment (metachris) [inline crates/rbuilder/src/live_builder/simulation/mod.rs:131]: why is all of this not needed anymore?

Comment (ZanCorDX) [reply crates/rbuilder/src/live_builder/simulation/mod.rs:131]: This is an old comment now we handle this on OrderReplacementManager.

**Review by metachris — APPROVED** [review]

File: crates/rbuilder/src/live_builder/order_input/txpool_fetcher.rs [hunk 3]
```diff
@@ -144,6 +145,7 @@ mod test {
         }
     }
 
+    #[ignore]
```
Comment (fiiiu) [inline crates/rbuilder/src/live_builder/order_input/txpool_fetcher.rs]: did this one sneak in or is this deliberate?

File: crates/rbuilder/src/live_builder/payload_events/mod.rs [hunk 4]
```diff
@@ -30,11 +30,14 @@ const NEW_PAYLOAD_RECV_TIMEOUT: Duration = SLOT_DURATION.saturating_mul(2);
...
 
+/// Data about a slot received from relays.
+/// Contains the important information needed to build and submit the block.
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct MevBoostSlotData {
+    /// The .data.payload_attributes.suggested_fee_recipient is replaced
     pub payload_attributes_event: PayloadAttributesEvent,
     pub suggested_gas_limit: u64,
-    /// List of relays that have this slot registered
+    /// List of relays that have this slot registered.
```
Comment (fiiiu) [inline crates/rbuilder/src/live_builder/payload_events/mod.rs]: this wording isn't great.. maybe "List of relays registered for this slot"? not sure about the semantics.

File: crates/rbuilder/src/mev_boost/mod.rs [hunk 5]
```diff
@@ -505,7 +510,7 @@ impl RelayClient {
         Ok(builder.send().await.map_err(RelayError::RequestError)?)
     }
 
-    #[allow(clippy::too_many_arguments)]
```
Comment (fiiiu) [inline crates/rbuilder/src/mev_boost/mod.rs:508]: why is this no longer needed? the amount of arguments didn't change

**Review by fiiiu — APPROVED** [review]
LGTM, left a couple of minor comments to check.

File: crates/rbuilder/src/live_builder/order_input/txpool_fetcher.rs (hunk 3 above)
Comment (ZanCorDX) [reply crates/rbuilder/src/live_builder/order_input/txpool_fetcher.rs]: Good catch! I have a problem with that test but on CI it works, never had the time to solve it.

File: crates/rbuilder/src/mev_boost/mod.rs (hunk 5 above)
Comment (ZanCorDX) [reply crates/rbuilder/src/mev_boost/mod.rs:508]: Probably some refactoring made it nicer and we forgot to remove it.

**Pull request merged**